gc.hpp

```diff
@@ -94,6 +104,9 @@ void mark(uint8_t d, GCObject *object)
   }
 }
 
+// The function which walks the stack, it requires an Iterator which
+// is responsible for the strategy that will be used to walk the stack.
+// PSA: C++ iterators are not as flexible as I thought and caoused more problems then it solved.
```

Comment:
That last sentence is intriguing! What happened?

[Also spelling: "caused".]

Reply:
It turns out c++ iterators work in a different way than most other programming languages for eg. java where they are a high lvevel abstraction and can be used fairly easy with all kinds of things. In c++ it took a lot of hacking to fit stackscanning to look like an iterator, which it looks like if you squint hard enough but I spend more time than I wanted on trining to debug issues arasing from this.

Reply:
Does it still cause problems? I guess I'm now wondering whether the comment is still relevant or not.

Reply:
It is working fine now, but if you look at the code, it is doing some strange non-standard things.